photon-client/src/integTest/scala/com/linkedin/photon/ml/cli/game/training/GameTrainingDriverIntegTest.scala

```diff
@@ -569,22 +557,14 @@ class GameTrainingDriverIntegTest extends SparkTestUtils with GameTestUtils with
    * @param expectedNumCoefficients Expected number of non-zero coefficients
    * @return True if the model is sane
    */
-  private def assertModelSane(path: Path, expectedNumCoefficients: Int, modelId: Option[String] = None): Unit = {
+  private def assertModelSane(path: Path, expectedNumCoefficients: Int): Unit = {
 
     val modelAvro = AvroUtils.readFromSingleAvro[BayesianLinearModelAvro](
       sc,
       path.toString,
       BayesianLinearModelAvro.getClassSchema.toString)
 
-    val model = modelId match {
-      case Some(id) =>
-        val m = modelAvro.find { m => m.getModelId.toString == id }
-        assertTrue(m.isDefined, s"Model id $id not found.")
-        m.get
-      case _ => modelAvro.head
-    }
-
-    assertEquals(model.getMeans.count(x => x.getValue != 0), expectedNumCoefficients)
+    assertEquals(modelAvro.head.getMeans.count(x => x.getValue != 0), expectedNumCoefficients)
```

Comment:
Does this work? The model in `head` position might not be the one that the test case expects. This led to some nondeterminism in the past.

Reply:
`assertModelSane` is only being called now for fixed effects and random effects that should have a uniform number of features across all models.

Reply:
Got it.